### PR TITLE
Helm Plugin - Add API

### PR DESCRIPTION
**Pull request by camilamacedo86**
Add the helm plugin to scaffold new APIs.
The call will be done in a follow-up. 

blocked by #3239

### CODE REVIEW

File: internal/plugins/helm/plugin.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	_ plugin.CreateAPIPluginGetter = Plugin{}
+)
+
+type Plugin struct {
+	createAPIPlugin
+}
+
+func (Plugin) Name() string                           { return pluginName }
+func (Plugin) Version() plugin.Version                { return plugin.Version{} }
+func (Plugin) SupportedProjectVersions() []string     { return nil }
```
Comment (estroz) [inline internal/plugins/helm/plugin.go:33]: The helm plugin itself should be added in another PR where we decide on a version, supported project versions, and name.

Comment (joelanford) [reply internal/plugins/helm/plugin.go:33]: If we go that route, we would need to export the `initPlugin` struct so that we could access it directly in the CLI.

I don't have a strong opinion on this though since `SupportedProjectVersions()` returning `nil` should mean this can't actually be used as is in a CLI (IIUC).

For `Name()`, we could return `""` for now.

Comment (camilamacedo86) [reply internal/plugins/helm/plugin.go:33]: Hi @estroz, 

The strategy is:
- Add only the plugins(without the config) 
- Then, when we have to init and create api for Ansible and Helm we will do 1 pr to call then in the new command.
- So, after that I see we following up with; the parse of the flags (domain/group/version), config and then cli. 

In this way, the version will be just decided when we set up the config which is out of the scope. I hope that it clarifies.

Comment (estroz) [reply internal/plugins/helm/plugin.go:33]: Ideally `Plugin` should be added in a separate PR and a `createAPIPlugin` field added here to keep the scope tight.

However you're right, my comment on name etc. shouldn't block this PR, since we should be deciding these defining attributes when we add the plugin to the new CLI.

Comment (camilamacedo86) [reply internal/plugins/helm/plugin.go:33]: Hi @estroz, we have the init and api. We will merge the init first and then, rebase this one with the master and it will be just the api as you suggested. See; https://github.com/operator-framework/operator-sdk/pull/3239.

I am looking for your ok/lgtm for we are able to move forward with it. ( init first -> rebase -> then just api )

Comment (estroz) [reply internal/plugins/helm/plugin.go:33]: That's fine I suppose.

File: internal/plugins/helm/api.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	fs.StringVar(&p.HelmChartRef, helmChart, "",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path).")
+	fs.StringVar(&p.HelmChartVersion, helmChartVersion, "",
+		"Specific version of the helm chart (default is latest version)")
+	fs.StringVar(&p.HelmChartRepo, helmChartRepo, "",
+		"Chart repository URL for the requested helm chart")
+}
+
+func (p *createAPIPlugin) InjectConfig(c *config.Config) {
+	c.Layout = plugin.KeyFor(Plugin{})
```
Comment (estroz) [inline internal/plugins/helm/api.go:82]: Only `Init` should do this.
```suggestion
```
There should probably be a follow-up in kb to do this automatically, otherwise every `Init` plugin must set the layout themselves.

Comment (camilamacedo86) [reply internal/plugins/helm/api.go:82]: good catcher.

**Comment by camilamacedo86** [comment]
Hi @estroz and @joelanford,

I applied some suggestions and did a few improvements. Let's me know what do you think now and if you see anything else that should be changed before we merge this one.

File: internal/plugins/helm/flags.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+const (
+	apiVersion       = "api-version"
+	kind             = "kind"
+	crdVersion       = "crd-version"
+	helmChart        = "helm-chart"
+	helmChartVersion = "helm-chart-version"
+	helmChartRepo    = "helm-chart-repo"
+)
+
+type APIFlags struct {
```
Comment (estroz) [inline internal/plugins/helm/flags.go:35]: These are only ever used in one place, so why not just have these fields in `createAPIPlugin`? At the very least this shouldn't be exported.

Comment (camilamacedo86) [reply internal/plugins/helm/flags.go:35]: We split the PR init/api. The flags are common for both and are used for them. See: #3239

File: internal/plugins/helm/api.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	}
+
+	if err := p.apiFlags.Validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *createAPIPlugin) Scaffold() error {
+	// Check if the project name is a valid namespace according to k8s
```
Comment (estroz) [inline internal/plugins/helm/api.go:88]:
```suggestion
```
